_docs/javascript-api.md

```diff
@@ -1561,6 +1582,35 @@ var Activity = Java.use("android.app.Activity");
 var activity = Java.cast(ptr("0x1234"), Activity);
 {% endhighlight %}
 
+**Note** Difference between Java.use() and Java.choose() 
+
+*use () :*
+Doest not give direct access to the objects at all. 
+Gives access to the object via the `this` operator.
+
+Let's take this as an example :
+
+{% highlight js %}
+Java.perform(function () {
+    var Activity = Java.use("gca.lc");
+    Activity.methodM1.overload('[B', 'java.lang.String').implementation = function (a, str) {
+        var retval = this.methodM1(a, str);
+        console.log("[*] return value4: "+retval);
+        return retval;
+    };
+    });
+{% endhighlight %}
+
+So in the above we are saying that :
+Give me a javascript wrapper for the class `lc` in the package `gca`. This is being referenced by `Activity` in our case. Now we say that whenever the method, `methodM1`, from `gca.lc` (which in this case also happens to be overloaded) gets called by any object (or from anywhere in the app), hook it and run my javascript function as defined above. Inside this javascript function, I am using the `this` operator to call methodM1 explicitly with the current object (which we have access to through the `this` operator). 
```

Comment:
- Let's call it `JavaScript` for consistency.
- "This is being referenced by `Activity`" – what does this mean?
- `this` is not an operator.